core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestUseConnectorTableScanPartitioning.java

```diff
@@ -0,0 +1,308 @@
+        tester.getQueryRunner().getNodePartitioningManager().addPartitioningProvider(
+                new CatalogName(MOCK_CATALOG),
+                new TestPartitioningProvider(new InMemoryNodeManager()));
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        closeAllRuntimeException(tester);
+        tester = null;
+    }
+
+    @Test
+    public void testEnablePlanWithTableNodePartitioning()
+    {
+        testPlanWithTableNodePartitioning(
+                ENABLE_PLAN_WITH_TABLE_NODE_PARTITIONING,
+                PARTITIONED_TABLE_HANDLE,
+                CONNECTOR_PARTITIONED_TABLE_HANDLE,
+                true);
+    }
+
+    @Test
+    public void testDisablePlanWithTableNodePartitioning()
+    {
+        testPlanWithTableNodePartitioning(
+                DISABLE_PLAN_WITH_TABLE_NODE_PARTITIONING,
+                PARTITIONED_TABLE_HANDLE,
+                CONNECTOR_PARTITIONED_TABLE_HANDLE,
+                false);
+    }
+
+    @Test
+    public void testTableScanWithoutConnectorPartitioning()
+    {
+        testPlanWithTableNodePartitioning(
+                ENABLE_PLAN_WITH_TABLE_NODE_PARTITIONING,
+                UNPARTITIONED_TABLE_HANDLE,
+                CONNECTOR_UNPARTITIONED_TABLE_HANDLE,
+                false);
+    }
+
+    @Test
+    public void testTableScanWithFixedConnectorPartitioning()
+    {
+        testPlanWithTableNodePartitioning(
+                DISABLE_PLAN_WITH_TABLE_NODE_PARTITIONING,
+                FIXED_PARTITIONED_TABLE_HANDLE,
+                CONNECTOR_FIXED_PARTITIONED_TABLE_HANDLE,
+                true);
+    }
+
+    private void testPlanWithTableNodePartitioning(
+            Session session,
+            TableHandle tableHandle,
+            ConnectorTableHandle connectorTableHandle,
+            boolean expectedEnabled)
+    {
+        tester.assertThat(new UseConnectorTableScanPartitioning(tester.getMetadata(), tester.getQueryRunner().getNodePartitioningManager()))
+                .on(p -> {
+                    Symbol a = p.symbol(COLUMN_A);
+                    Symbol b = p.symbol(COLUMN_B);
+                    return p.tableScan(tableHandle,
+                            ImmutableList.of(a, b),
+                            ImmutableMap.of(a, COLUMN_HANDLE_A, b, COLUMN_HANDLE_B));
+                })
+                .withSession(session)
+                .matches(
+                        tableScan(
+                                equalTo(connectorTableHandle),
+                                TupleDomain.all(),
+                                ImmutableMap.of(
+                                        "A", equalTo(COLUMN_HANDLE_A),
+                                        "B", equalTo(COLUMN_HANDLE_B)))
+                                .with(planWithTableNodePartitioning(expectedEnabled)));
+    }
+
+    private Matcher planWithTableNodePartitioning(boolean enabled)
```

Comment:
Why not add `Optional<Boolean> useConnectorTablePartitioning` to TableScanMatcher ?

Reply:
It's just single use case, so I don't think it's worth it. When more cases arrive, we could think about it.